backend/sql/initial_schema.sql

```diff
@@ -0,0 +1,100 @@
+create table project
+(
+    project_id  bigserial     not null primary key,
+    name        varchar(256)  not null unique,
+    description varchar(1000) null
+);
+
+create table test
+(
+    test_id bigserial    not null primary key,
+    name    varchar(256) not null unique
+);
+
+create table subtest
+(
+    subtest_id      bigserial      not null primary key,
+    test_id         bigint         not null,
+    name            varchar(256)   not null unique,
+    -- Method
+    meth            varchar(10000) null,
+    confluence_link varchar(1000)  null,
+    constraint belongs_to_test
+        foreign key (test_id) references test (test_id)
+            on update cascade
+            on delete cascade
+);
+
+-- Corresponds to an EXCEL file
+create table experiment
+(
+    experiment_id bigserial                not null primary key,
+    project_id    bigint                   not null,
+    subtest_id    bigint                   not null,
+    -- This is input from the user
+    name          varchar(256)             null,
+    meta          varchar(1000)            null,
+    -- This is fallback name to show if name is not specified
+    raw_file_name varchar(256)             not null,
+    upload_date   timestamp with time zone not null,
+
+    constraint belongs_to_project
+        foreign key (project_id) references project (project_id)
+            on update cascade
+            on delete cascade,
+
+    constraint has_subtest
+        foreign key (subtest_id) references subtest (subtest_id)
+            on update cascade
+            -- Prevent removing subtest if there is an experiment having such subtest
+            on delete no action
+);
+
+create table measurement
+(
+    measurement_id bigserial        not null primary key,
+    experiment_id  bigint           not null,
+    -- Either compound id or name
+    compound_id    varchar(256)     not null,
+    concentration  double precision not null,
+    signal         double precision not null,
+
+    constraint belongs_to_experiment
+        foreign key (experiment_id) references experiment (experiment_id)
+            on delete cascade
+            on update cascade
+);
+
+-- We can remove this table if we don't need anything apart from IC50
+create table analysis
+(
+    analysis_id bigserial     not null primary key,
+    description varchar(1000) null,
+    -- Analysis-specific parameters
+    parameters  json          null
+);
+
+create table analysis_outcome
+(
+    outcome_id    bigserial          not null primary key,
+    analysis_id   bigint             not null,
+    experiment_id bigint             not null,
+    compound_id   bigint             not null,
+    -- Maybe JSON better?
+    result        double precision[] not null,
```

Comment:
Well, result is an interesting beast. TL;DR: I agree that we should track IC50 result in `analysis_outcome` as listed here, but the reason why is non-trivial.

First of all, let's talk about the actual result of conducting of an experiment as far as our knowledge about effectiveness of drugs is concerned.

To begin with, there is no "absolute correct IC50". To determine "absolute IC50" we would need to admit drugs of various concentrations to every single biological unit that has the biological function the drug targets and conduct precise measurements. IC50 itself is a statistical / probabilistic value. By anlysing the data we have, however, we can do analyses that will make us be increasingly sure that IC50 value for a drug we have is predicting what will happen in real world after we admit that drug.

So... when we run an experiment, we already have some IC50 analyses done. Depending on how good we are at choosing only good data for said analyses, we may or may not have a pool of high-quality data and we have some mean IC50 and a value for some deviation (scientists propose standard deviation, but we need to be sure that the distribution of concentration readouts is normal, perhaps in logarithmed fashinon) in the data set. After the experiment is conducted, we add more (hopefully high-quality) data to the pool, making mean IC50 more precise. So the result of our analysis is actually updated tuple of (mean, deviation) for the *compound*.

If we insist on tracking results of a particular analysis, we must to do the following:

  1. calculate IC50 based on the points submitted for analysis
  2. determine where in the distribution lies this calculated value
  
Sadly, point (2) moves over time and it would be a hassle to update it every time a compound gets new high-quality data imported into the database, so *it's OK to just track IC50* of the analysus, but it's *not why we're storing `analysis_outcome`s or analyses in general*.